### PR TITLE
[Bug 18293] Use MCObjectHandle for button menu

### PR DESCRIPTION
**Pull request by montegoulding**

### CODE REVIEW

File: engine/src/button.cpp [hunk 1]
```diff
@@ -412,8 +412,13 @@ MCButton::~MCButton()
 
 void MCButton::removelink(MCObject *optr)
 {
-	if (optr == menu)
+	if (menu.IsValid() && optr == menu)
+    {
+        MCValueRelease(menuname);
+        menuname = MCValueRetain(kMCEmptyName);
```
Comment (livecodeali) [inline engine/src/button.cpp]: You can just use `MCValueAssign` here

File: engine/src/button.cpp [hunk 2]
```diff
@@ -2603,11 +2613,14 @@ Boolean MCButton::findmenu(bool p_just_for_accel)
 				MCU_set_rect(trect, 0, 0, rect.width, height + 4);
 				trect = MCU_reduce_rect(trect, MClook == LF_MOTIF ? DEFAULT_BORDER : 1);
 				fptr->setrect(trect);
-				/* UNCHECKED */ MCStackSecurityCreateStack(menu);
-
-				menu->setparent(this);
-				menu->createmenu(fptr, rect.width, height + 4);
-				MCdispatcher->appendpanel(menu);
+                MCStack * t_menu;
+				/* UNCHECKED */ MCStackSecurityCreateStack(t_menu);
```
Comment (livecodeali) [inline engine/src/button.cpp]: _dons @peter-b hat_ This could probably `return False` here if creating the stack fails, couldn't it?

Comment (montegoulding) [inline engine/src/button.cpp]: 🎩 ?

**Comment by montegoulding** [comment]
Just hit this crash again. Perhaps this should target develop-8.1?

**Comment by peter-b** [comment]
Hi Monte, can you please file an appropriate bug report and add a release note to this PR please?

**Comment by peter-b** [comment]
Also @livecodeali says that he has some code that could easily be turned into a test for this defect.

**Comment by montegoulding** [comment]
I'll look at this again when https://github.com/livecode/livecode/pull/4430 is merged as that will make it pretty.

File: engine/src/stack.cpp [hunk 3]
```diff
@@ -1379,7 +1373,10 @@ Boolean MCStack::del(bool p_check_flag)
 {
     if (!isdeletable(true))
 	   return False;
-	
+    
+    MCscreen->ungrabpointer();
+    MCdispatcher->removemenu();
```
Comment (montegoulding) [inline engine/src/stack.cpp:1380]: I'm not positive these are reasonable... I suspect I need to add a way to ungrab if it's the stack window that is grabbed and the same with removemenu... on the other hand while a popup grabbed are we really deleting stacks?

File: engine/src/button.cpp [hunk 4]
```diff
@@ -412,8 +410,11 @@ MCButton::~MCButton()
 
 void MCButton::removelink(MCObject *optr)
 {
-	if (optr == menu)
-		menu = NULL;
+	if (menu.IsValid() && optr == menu)
```
Comment (livecodefraser) [inline engine/src/button.cpp]: This should probably be "IsBound()" here - we want to check if the handle currently points at something, not whether that something still exists or not.

Comment (montegoulding) [reply engine/src/button.cpp]: OK, thanks

Comment (montegoulding) [reply engine/src/button.cpp]: done

**Comment by livecodefraser** [comment]
@livecode-vulcan review okay 3e85689

**Comment by livecode-vulcan** [comment]
💙 review by @livecodefraser ok 3e8568928934050a5f4cc8ee5c78469b47289746

**Comment by livecode-vulcan** [comment]
😎 test success 3e8568928934050a5f4cc8ee5c78469b47289746
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/226)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/230)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/226)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/226)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/230)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/226)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/226)

**Pull request merged**